File: Data Structures & Algorithms/valid-binary-search-tree/submission-7.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    List<Integer> res;
    private void bfs(TreeNode node) {
        if (node == null) {
            return;
        }
        bfs(node.left);
        res.add(node.val);
        bfs(node.right);
    }
    public boolean isValidBST(TreeNode root) {
        res = new ArrayList<>();
        int prev = -1001;
        bfs(root);
        for (int num: res) {
            if (num <= prev) {
                return false;
            }
            prev = num;
        }
        return true;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-7.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        res = []
        total = 1
        for num in nums:
            res.append(total)
            total*= num
        
        total = 1
        for i in range(len(nums)-1, -1, -1):
            res[i] *= total
            total *= nums[i]
        return res
        

        